### PR TITLE
Fix build errors

### PR DESCRIPTION
**Pull request by cond0r**
- revert deleted files from api folder (cannot run `yarn build` without it yet)
- revert / downgrade dependencies (the generated site  with `yarn build`, doesn't render images, breaks fetcher + links after updated dependencies)
- modified queries & parts of code that remained from Shopify provider and caused errors,  changed to fit Saleor API like: product sortBy, product filter by collection 
- fix error after cart item remove 
- fix other several build errors

https://commerce-2qmuftbb7-cond0r.vercel.app

### CODE REVIEW

**Comment by zaiste** [comment]
@cond0r We are still in the process of adapting Saleor for Next.js Commerce. Given our internal roadmap it's more a work-in-progress than a final stage. Thanks for the suggestions. It seems most of them can be integrated as-is. Would you mind rebasing it?

File: components/auth/SignUpView.tsx [hunk 1]
```diff
@@ -34,9 +34,11 @@ const SignUpView: FC<Props> = () => {
       setMessage('')
       await signup({
         email,
-        firstName,
-        lastName,
         password,
+        metadata: [
```
Comment (royderks) [inline components/auth/SignUpView.tsx]: Unfortunately, we cannot make this changes in this branch. They should be added to the upstream branch of `commerce:master`

File: package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "nextjs-commerce",
```
Comment (royderks) [inline package.json:2]: Could you also revert these to match with `commerce:master` and create a seperate branch for the package updates?

**Comment by royderks** [comment]
Great PR so far @cond0r, seems very useful 🥇.  Could you have a look at my comments? Let me know if I can help :)

**Comment by cond0r** [comment]
@zaiste it's ok, actually i didn't saw the other pull request, with the changes that covered some things when i made it. Makes more sense now since there are things to do on Saleor too. So the other fixes was about reverting package.json dependencies, revert files in api folder so it can build. Maybe just consider it.

**Comment by zaiste** [comment]
@cond0r for the revert in `package.json`, did you also have issues with the build process?

**Comment by zaiste** [comment]
@cond0r Perfect! Could we remove the files in `framework/saleor/api/*` from this PR? I've added the stubs without directories. Also, it seems that only Next.js itself needs to be downgraded, all other dependencies can be left at the most recent versions. What do you think?

**Comment by zaiste** [comment]
@cond0r would it be possible to squash the last three commits into a single one?

**Pull request closed without merging**